src/base/PositionManager.sol

```
@@ -66,19 +66,21 @@ contract PositionManager is IPositionManager, Multicall, PositionNFT, PermitERC2
 
     function decreaseLiquidity(DecreaseLiquidityParams calldata params_) external override payable mayInteract(params_.pool, params_.tokenId) nonReentrant {
         uint256 curPos = positions[params_.tokenId].lpTokens[params_.index];
-        require(params_.lpTokens <= curPos, "PM:DL:INSUF_LP_BAL");
+        require(params_.lpTokens != 0 && params_.lpTokens <= curPos, "PM:DL:INSUF_LP_BAL");
```

I don't think we should check for `params_.lpTokens != 0` and leave it as an user / dapp responsibility? Any special reason you included here?

---

With the old implementation, user could call with 0 LP.  `removeCollateral` and `removeQuoteToken` would silently do nothing, so user would pay the gas without withdrawing anything.

I'll remove this check.  Calling with 0 LP should cause `removeQuoteToken` to revert.